File: past-competitions/2018/src/brett-cpp/Prob09.c
#include <string.h>
#include <stdio.h>
#include <stdlib.h>
#include <math.h>

int main(int argc, char *argv[])
{
	// open file for reading
	FILE *inputFile = fopen("Prob09.in.txt", "r");
	int testCases;

	// read number of test cases
	fscanf(inputFile, "%d\n", &testCases);
	
	int a, b, c, temp;

	// execute test cases
	while(testCases > 0){
		// read the initial a and b values
		fscanf(inputFile, "%d,%d\n", &a, &b);

		// swap them if needed so a is larger
		if(a < b){
			temp = a;
			a = b;
			b = temp;
		}

		do{
			// subtract b from a (into c)
			c = a - b;
			// print the equation
			printf("%d-%d=%d\n", a, b, c);
			if(c != 0){
				// if c isn't 0, set us up for the next round
				a = b;
				b = c;
				// swap a and b if needed
				if(a < b){
					temp = a;
					a = b;
					b = temp;
				}
			}
		}while(c != 0);

		// if a and b are 1, the original numbers are coprime
		if(a == 1 && b == 1){
			printf("COPRIME\n");
		}
		// otherwise they ain't
		else{
			printf("NOT COPRIME\n");
		}

		testCases = testCases - 1;
	}

	// close the file
	fclose(inputFile);
}

File: past-competitions/2018/src/brett-python/Prob09.py
# Open the input file
with open("Prob09.in.txt", "rt") as inputFile:
    # Read the number of test cases (trim out the newline)
    cases = int(inputFile.readline().replace("\n", ""))
    # For each test case
    for caseNum in range(cases):
        # Read the line
        line = inputFile.readline().replace("\n", "")
        # split into digits
        numbers = line.split(",")
        a = int(numbers[0])
        b = int(numbers[1])
        c = 100
        swap = 0
        # continue process until c is 0
        while c != 0:
            # swap a and b if needed
            if a < b:
                swap = a
                a = b
                b = swap
            # subtract and print equation
            c = a - b
            print(str(a) + "-" + str(b) + "=" + str(c))
            # shift values
            a = b
            b = c
        # print result
        if a == 1:
            print("COPRIME")
        else:
            print("NOT COPRIME")
